src/core/common/message.cpp

```diff
@@ -193,18 +193,17 @@ uint16_t MessagePool::GetTotalBufferCount(void) const
 //---------------------------------------------------------------------------------------------------------------------
 // Message::Settings
 
-const Message::Settings Message::Settings::kDefault(Message::kWithLinkSecurity, Message::kPriorityNormal);
+const otMessageSettings Message::Settings::kDefault = {kWithLinkSecurity, kPriorityNormal};
 
 Message::Settings::Settings(LinkSecurityMode aSecurityMode, Priority aPriority)
-    : mLinkSecurityEnabled(aSecurityMode == kWithLinkSecurity)
-    , mPriority(aPriority)
```

Comment:
```cpp
 : otMessageSettings{aSecurityMode, aPriority}
```

Reply:
We can change to this but I slightly prefer to keep it as is.
- This relies on aggregate initialization of C struct (since `otMessageSettings` is an "extern C" struct, we cannot have a constructor for it). 
- With this we need to make sure the order of parameter in `{}` are correct (complier won't actually check/verify this for us). With the assignments it will always be correct.